### PR TITLE
SFD-114 - Add playwright tests into automated testing

### PR DESCRIPTION
**Pull request by mgriffin-scottlogic**
Move existing playwright workflow
Update with local actions and Python specific requirements
Made part of the PR process for now, see if it causes any issues.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -21,3 +21,13 @@ jobs:
         uses: ./.github/actions/set-up-environment
       - name: Run tests
         run: npm test
+  playwright_tests:
+    needs: unit_tests
```
Comment (jantoun-scottlogic) [inline .github/workflows/pull-request.yml:25]: Should we run the Playwright tests in parallel with the unit tests?

Comment (jantoun-scottlogic) [reply .github/workflows/pull-request.yml:25]:
```suggestion
    needs: build
```

Comment (mgriffin-scottlogic) [reply .github/workflows/pull-request.yml]: Yeah, happy to go with that. My initial thought was they follow on from each other but there's nothing worse than having to run tests multiple times just to find all of the failures.

File: .github/actions/run-playwright/action.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      run: npm run playwright-install
+      shell: bash
+    - name: Run Playwright tests
+      run: npm run playwright-test
+      shell: bash
+    - uses: actions/upload-artifact@v4
+      if: always()
+      with:
+        name: playwright-report
+        path: playwright/playwright-report/
```
[the remark this answers is not in the document]
Comment (mgriffin-scottlogic) [reply .github/actions/run-playwright/action.yml:20]: Yes, I could drop this section for now, I couldn't see a simple way to activate the reporting. It's possible that like the leftover .spec.ts file, this was intended for use with the javascript version of the library.

[the remark this answers is not in the document]
Comment (jantoun-scottlogic) [reply .github/actions/run-playwright/action.yml]: Might be that the path just needs to be changed. In the [Playwright docs example](https://playwright.dev/python/docs/ci-intro#on-pushpull_request) the path is `test-results/`.

[the remark this answers is not in the document]
Comment (jantoun-scottlogic) [reply .github/actions/run-playwright/action.yml]: Does it upload the report when the tests fail? Playwright may not be producing a report by default if the tests pass.

[the remark this answers is not in the document]
Comment (mgriffin-scottlogic) [reply .github/actions/run-playwright/action.yml]: It's definitely never generated anything by default, had plenty of failing runs to find one way or another :)

[the remark this answers is not in the document]
Comment (jantoun-scottlogic) [reply .github/actions/run-playwright/action.yml]: As discussed we can merge without the artifacts being uploaded for now and look into this.

**Review by jantoun-scottlogic — APPROVED** [review]

**Pull request merged**